### PR TITLE
flake8 modeling

### PR DESCRIPTION
**Pull request by WilliamJamieson**
### Description

Just ran `flake8 .` on the modeling module and fixed all the errors that uncovered in a similar process to #13087.

There were lots of issues because the flake8 CI does not check for very much. Maybe more checks should be enabled?

### Checklist for package maintainer(s)

This checklist is meant to remind the package maintainer(s) who will review this pull request of some common things to look for. This list is not exhaustive.

- [x] Do the proposed changes actually accomplish desired goals?
- [x] Do the proposed changes follow the [Astropy coding guidelines](https://docs.astropy.org/en/latest/development/codeguide.html)?
- [x] Are tests added/updated as required? If so, do they follow the [Astropy testing guidelines](https://docs.astropy.org/en/latest/development/testguide.html)?
- [x] Are docs added/updated as required? If so, do they follow the [Astropy documentation guidelines](https://docs.astropy.org/en/latest/development/docguide.html#astropy-documentation-rules-and-guidelines)?
- [ ] Is rebase and/or squash necessary? If so, please provide the author with appropriate instructions. Also see ["When to rebase and squash commits"](https://docs.astropy.org/en/latest/development/when_to_rebase.html).
- [ ] Did the CI pass? If no, are the failures related? If you need to run daily and weekly cron jobs as part of the PR, please apply the `Extra CI` label.
- [x] Is a change log needed? If yes, did the change log check pass? If no, add the `no-changelog-entry-needed` label. If this is a manual backport, use the `skip-changelog-checks` label unless special changelog handling is necessary.
- [x] Is this a big PR that makes a "What's new?" entry worthwhile and if so, is (1) a "what's new" entry included in this PR and (2) the "whatsnew-needed" label applied?
- [x] Is a milestone set? Milestone must be set but `astropy-bot` check might be missing; do not let the green checkmark fool you.
- [x] At the time of adding the milestone, if the milestone set requires a backport to release branch(es), apply the appropriate `backport-X.Y.x` label(s) *before* merge.

### CODE REVIEW

**Comment by pllim** [comment]
> Maybe more checks should be enabled?

We decided against it a long time ago due to the amount of accumulated tech debt in the code base. In particular, one should not touch `io.fits`.

File: astropy/modeling/bounding_box.py
Comment (pllim) [inline astropy/modeling/bounding_box.py:1422]: Maybe out of scope but I can't help but notice in this diff... is the exclamation mark really necessary? 
```suggestion
                             "(can be passed as part of bounding_box argument)")
```

Comment (WilliamJamieson) [reply astropy/modeling/bounding_box.py]: Why not!. See changes in 953ebe4.

File: astropy/modeling/tests/test_fitters.py
Comment (pllim) [inline astropy/modeling/tests/test_fitters.py:471]: To be consistent?
```suggestion
                                r'clipping to bounds')
```

File: astropy/modeling/tests/test_fitters.py
Comment (pllim) [inline astropy/modeling/tests/test_fitters.py:490]: To be consistent?
```suggestion
                                r'clipping to bounds')
```

File: astropy/modeling/tests/test_input.py
Comment (pllim) [inline astropy/modeling/tests/test_input.py:152]:
```suggestion
            pfit(p1, self.x1, y1)
```

File: astropy/modeling/tests/test_input.py
Comment (pllim) [inline astropy/modeling/tests/test_input.py:504]:
```suggestion
            t(np.arange(5) * 100)
```

File: astropy/modeling/tests/test_model_sets.py
Comment (pllim) [inline astropy/modeling/tests/test_model_sets.py:306]:
```suggestion
    fitter(model, x, y.T)
```

File: astropy/modeling/tests/test_model_sets.py
Comment (pllim) [inline astropy/modeling/tests/test_model_sets.py:310]:
```suggestion
    fitter(model, x, x, y)
```

File: astropy/modeling/tests/test_model_sets.py
Comment (pllim) [inline astropy/modeling/tests/test_model_sets.py:316]:
```suggestion
        Polynomial1D(1, n_models=2, model_set_axis=1) | Polynomial1D(1, n_models=2, model_set_axis=0)  # noqa
```

File: astropy/modeling/tests/test_parameters.py
Comment (pllim) [inline astropy/modeling/tests/test_parameters.py:472]: I would prefer `\` is gotten rid of using parenthesis.

File: astropy/modeling/tests/test_parameters.py
Comment (pllim) [inline astropy/modeling/tests/test_parameters.py:1051]:
```suggestion
            TParModel(coeff=[[1, 2], [3, 4]], e=4, model_set_axis=0)
```

File: astropy/modeling/tests/test_parameters.py
Comment (pllim) [inline astropy/modeling/tests/test_parameters.py:1059]:
```suggestion
            TParModel(np.array([[1, 2], [3, 4]]), (1, 1, 11),
```

File: astropy/modeling/tests/test_quantities_evaluation.py
Comment (pllim) [inline astropy/modeling/tests/test_quantities_evaluation.py:319]: And here: `\`

File: astropy/modeling/tests/test_quantities_parameters.py
Comment (pllim) [inline astropy/modeling/tests/test_quantities_parameters.py:335]:
```suggestion
    Pix2Sky_TAN()
```

File: astropy/modeling/tests/test_rotations.py
Comment (pllim) [inline astropy/modeling/tests/test_rotations.py:248]:
```suggestion
```

File: astropy/modeling/core.py
Comment (WilliamJamieson) [reply astropy/modeling/core.py:3172]: Yes this is fixing a missing import.

Comment (pllim) [reply astropy/modeling/core.py:3172]: If the PR that introduced this bug is in 5.0.x, we need to backport, otherwise it can stay here. FYI.

File: astropy/modeling/tests/test_fitters.py
Comment (WilliamJamieson) [reply astropy/modeling/tests/test_fitters.py:624]: Agreed.

File: astropy/modeling/core.py
Comment (WilliamJamieson) [reply astropy/modeling/core.py:3172]: This is a bug on the 5.0.4 tag.

File: astropy/modeling/tests/test_quantities_evaluation.py
Comment (WilliamJamieson) [reply astropy/modeling/tests/test_quantities_evaluation.py:319]: Done in 10d5232

File: astropy/modeling/tests/test_polynomial.py
Comment (WilliamJamieson) [reply astropy/modeling/tests/test_polynomial.py]: Done in 10d5232

File: astropy/modeling/tests/test_parameters.py
Comment (WilliamJamieson) [reply astropy/modeling/tests/test_parameters.py]: Done in 10d5232

File: astropy/modeling/tests/test_bounding_box.py
Comment (WilliamJamieson) [reply astropy/modeling/tests/test_bounding_box.py]: Done in 10d5232

File: astropy/modeling/core.py
Comment (WilliamJamieson) [reply astropy/modeling/core.py]: Done in 10d5232

File: astropy/modeling/core.py
Comment (WilliamJamieson) [reply astropy/modeling/core.py]: Went back to `lambda` in 10d5232

File: astropy/modeling/tests/test_fitters.py
Comment (WilliamJamieson) [reply astropy/modeling/tests/test_fitters.py]: Refactored in 40c2aa5.

File: astropy/modeling/core.py
Comment (WilliamJamieson) [reply astropy/modeling/core.py:3172]: See #13140.

**Comment by WilliamJamieson** [comment]
Note #13140 should be merged before this PR followed by rebasing this PR on main.

**Comment by pllim** [comment]
#13140 is merged. Please rebase. Thanks!

File: astropy/modeling/core.py
Comment (nstarman) [inline astropy/modeling/core.py:3438]: targeted ``noqa`` .  E731?

File: astropy/modeling/core.py
Comment (nstarman) [inline astropy/modeling/core.py:3718]: Did you run flake8 with line length 100? I think the old line was less than 100 characters.

File: astropy/modeling/fitting.py
Comment (nstarman) [inline astropy/modeling/fitting.py:80]: A good opportunity to replace the `\`

```
ret_str += (fstring.replace(' '*len(param), param, 1)
            .format(repr(np.round(row[:i+1], round_val))[7:-2]))
```

File: astropy/modeling/powerlaws.py
Comment (nstarman) [inline astropy/modeling/powerlaws.py:308]: If it's just one line is line length < 100 ?

**Review by nstarman — COMMENTED** [review]
This looks really good. A few comments.

1. There are a number of blanket ``noqa``. It would be great to make them targeted to a specific error. I only highlighted a few.
2. Many lines look like they could be condensed to <100 characters but flake8 doesn't remove the line continuation `\`. Now seems a good opportunity to remove the `\` and rerun flake8 to make multi-lines where appropriate and sub-100 character lines where possible.

File: astropy/modeling/core.py
Comment (WilliamJamieson) [reply astropy/modeling/core.py:3718]: Its 104 characters unfortuately

File: astropy/modeling/powerlaws.py
Comment (WilliamJamieson) [reply astropy/modeling/powerlaws.py:308]: It is weird that it was line wrapped at all. I just corrected the indentation error without really considering what was going on.

File: astropy/modeling/core.py
Comment (nstarman) [reply astropy/modeling/core.py:3718]: Oh well. externalities of a consistent style format.

File: astropy/modeling/core.py
Comment (WilliamJamieson) [reply astropy/modeling/core.py]: See 84b3922

File: astropy/modeling/fitting.py
Comment (WilliamJamieson) [reply astropy/modeling/fitting.py]: See 84b3922

File: astropy/modeling/powerlaws.py
Comment (WilliamJamieson) [reply astropy/modeling/powerlaws.py]: See 84b3922

**Review by pllim — APPROVED** [review]
This is good enough clean up for now. There is always more we can do but this PR doesn't have to be perfect.

I'll let @nstarman have one last look but this LGTM. Thanks!

**Review by nstarman — APPROVED** [review]
Agreed with @pllim.
As you say @WilliamJamieson, it would be nice to add more checks to the CI.

**Comment by nstarman** [comment]
@WilliamJamieson, squash?

**Pull request merged**
